Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -164,6 +165,22 @@ void ImageLoader::clearImageWithoutConsideringPendingLoadEvent()
         imageResource->resetAnimation();
 }
 
+// https://html.spec.whatwg.org/multipage/images.html#updating-the-image-data:list-of-available-images
+bool ImageLoader::canReuseFromListOfAvailableImages(const CachedResourceRequest& request, const String& crossOriginAttribute)
+{
+    CachedResource* resource = MemoryCache::singleton().resourceForRequest(request.resourceRequest(), document().page()->sessionID());
+    if (!resource || resource->stillNeedsLoad() || resource->isPreloaded())
+        return false;
+
+    if (!crossOriginAttribute.isNull() && !document().securityOrigin().isSameOriginAs(*resource->origin()))
+        return false;
+
+    if (resource->options().credentials == FetchOptions::Credentials::SameOrigin && (crossOriginAttribute.isNull() || equalLettersIgnoringASCIICase(crossOriginAttribute, "use-credentials"_s)))
+        return false;
+
+    return true;
```

Comment:
Should it be false when `resource->options().credentials == FetchOptions::Credentials::Include` and `crossOriginAttribute` is anonymous?

Reply:
You are right, based on what is in https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attribute it should be added too. In case `resource->options().credentials == FetchOptions::Credentials::Include`, `crossOriginAttribute` can't be null or different from `use-credentials`. I will add it.

Reply:
It seems that by default `resource->options().credentials` is set to `FetchOptions::Credentials::Include`:
https://github.com/WebKit/WebKit/blob/main/Source/WebCore/loader/ImageLoader.cpp#L187
https://github.com/WebKit/WebKit/blob/main/Source/WebCore/loader/cache/CachedResourceLoader.cpp#L1793

If the `crossorigin` attribute is not specified, `resource->options().credentials` should be set to `FetchOptions::Credentials::Omit` in: 
https://github.com/WebKit/WebKit/blob/main/Source/WebCore/loader/CrossOriginAccessControl.cpp#L162

In other case, we have situation that:
`resource->options().credentials == FetchOptions::Credentials::Include` and `crossOriginAttribute` is null

WDYT?

Reply:
If I set `FetchOptions::Credentials::Omit` for `resource->options().credentials`  in case of `crossOriginAttribute` is null than few tests fail. The reason for fails is that tests are expected to include credentials in the request of the resource.

Reply:
I think the problem is, when we decide if there is a match in `crossorigin` attribute of request and cached resource,
we only check if credentials mode (https://fetch.spec.whatwg.org/#concept-request-credentials-mode) is expected, but not mode (https://fetch.spec.whatwg.org/#concept-request-mode, `FetchOptions::mode`). According to spec of creating a core request (https://html.spec.whatwg.org/multipage/urls-and-fetching.html#create-a-potential-cors-request), `crossorigin` attribute could affect both mode and credentials mode. 

For example, in the case where img element's `crossorigin` attribute is not specified, potential request credentials mode should be 'include' and mode should be 'no-cors'. My understanding is if the cached resource's credentials mode is 'include' and mode is 'cors', it's a mismatch.

Reply:
Ok, now I see your point. Maybe I should also add tests similar to `LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/image-loading-lazy-use-list-of-available-images.html` but add options with different values for `crossorigin`.